src/replicator.js

```diff
@@ -0,0 +1,297 @@
+const EventEmitter = require('events')
+const OrbitDB = require('orbit-db')
+const Pubsub = require('orbit-db-pubsub')
+const AccessControllers = require('orbit-db-access-controllers')
+const resolveDID = require('did-resolver').default
+const {
+  LegacyIPFS3BoxAccessController,
+  ThreadAccessController,
+  ModeratorAccessController
+} = require('3box-orbitdb-plugins')
+AccessControllers.addAccessController({ AccessController: LegacyIPFS3BoxAccessController })
+AccessControllers.addAccessController({ AccessController: ThreadAccessController })
+AccessControllers.addAccessController({ AccessController: ModeratorAccessController })
+const config = require('./config')
+
+const PINNING_NODE = config.pinning_node
+const PINNING_ROOM = config.pinning_room
+const ORBITDB_OPTS = config.orbitdb_options
+const ODB_STORE_OPTS = {
+  ...ORBITDB_OPTS,
+  accessController: {
+    type: 'legacy-ipfs-3box',
+    skipManifest: true
+  }
+}
+const rootEntryTypes = {
+  SPACE: 'space',
+  ADDRESS_LINK: 'address-link',
+  AUTH_DATA: 'auth-data'
+}
+
+const pinDID = did => {
+  if (!did) return
+  // We resolve the DID in order to pin the ipfs object
+  try {
+    resolveDID(did)
+    // if this throws it's not a DID
+  } catch (e) {}
+}
+
+class Replicator {
```

Comment:
Well yeah, I think passing in the orbitdb instance is probably the way to go. The reason I put the store creation in here is because the replicator needs to be able to create the stores when it does the syncing. So I though it made sense for all db creation to happen here. And yeah the idea is to also add thread creation here similar to `addKVStore`.